x/stakeibc/keeper/unbonding_records.go

```diff
@@ -235,6 +245,93 @@ func (k Keeper) GetUnbondingICAMessages(
 	return msgs, unbondings, nil
 }
 
+// In the event that the number of generated undelegate messages exceeds the batch size,
+// reduce the number of messages by dividing any excess amongst proportionally based on
+// the remaining delegation
+// This will no longer be necessary after undelegations to 32+ validators is supported
+func (k Keeper) ConsolidateUnbondingMessages(
+	totalUnbondAmount sdkmath.Int,
+	initialUnbondings []*types.SplitDelegation,
+	unbondCapacities []ValidatorUnbondCapacity,
+	batchSize int,
+) (finalUnbondings []*types.SplitDelegation, err error) {
+	// Grab the first {batch_size} number of messages from the list
+	// This will consist of the validators with the most capacity
+	unbondingsCapped := initialUnbondings[:batchSize]
+
+	// Calculate the amount that was initially meant to be unbonded from that batch,
+	// and determine the remainder that needs to be redistributed
+	unbondAmountFromBatch := sdkmath.ZeroInt()
+	initialUnbondingsMap := map[string]sdkmath.Int{}
```

Comment:
Just pushed a new commit with some new names. Lemme know if it helps

For this particular variable, I renamed it to `initialUnbondAmountFromBatchByVal` - although, mentioning the batch is probably not necessary since this is just a map of the initial unbondings and we'll never lookup the vals outside of the batch

Reply:
The reason I was suggesting capped set be part somehow is that if we iterate over initialUnbondings or iterate over the capacities we would get validators outside the capped set.  In fact we do some of these types of iterations but with an if clause inside the loop which limits to validators inside the capped set.  We don't have to call it capped set if we don't want to but somehow make it clear it is a limited amount of the validators since that is one of the ways I got confused reading through the code the first time wondering if that was intended or a mistake to only have values for some but not all validators.

Reply:
does fromBatch clear it up though? Instead of capped?